### ISSUE TITLE
BUG: Backlinks are empty for Excalidraw pictures

### ISSUE THREAD

**Issue by koenbunders** [zsviczian/obsidian-excalidraw-plugin#2018]
### Have you searched for existing issues (including closed ones)?

- [X] Yes, I have searched the existing issues.

### Does this bug persist in a new vault with only Excalidraw installed?

- [X] Yes, I have verified the issue persists.

### Your environment

SYSTEM INFO:
	Obsidian version: v1.7.1
	Installer version: v1.4.16
	Operating system: Darwin Kernel Version 23.6.0: Mon Jul 29 21:14:30 PDT 2024; root:xnu-10063.141.2~1/RELEASE_ARM64_T6000 23.6.0
	Login status: logged in
	Language: en
	Catalyst license: insider
	Insider build toggle: on
	Live preview: on
	Base theme: light
	Community theme: Minimal v7.7.18
	Snippets enabled: 2
	Restricted mode: off
	Plugins installed: 42
	Plugins enabled: 31
		1: Calendar v1.5.10
		2: Commander v0.5.1
		3: Dataview v0.5.67
		4: Excalidraw v2.4.3
		5: ExcaliBrain v0.2.15
		6: Longform v2.0.7
		7: Natural Language Dates v0.6.2
		8: Note Refactor v1.8.2
		9: Outliner v4.8.1
		10: Paste URL into selection v1.7.0
		11: Periodic Notes v0.0.17
		12: Plugin Update Tracker v1.5.2
		13: QuickAdd v1.11.1
		14: Remember cursor position v1.0.9
		15: Sortable v0.3.1
		16: Style Settings v1.0.9
		17: Tag Wrangler v0.6.1
		18: Templater v2.7.1
		19: Workspaces Plus v0.3.3
		20: Minimal Theme Settings v8.1.1
		21: Supercharged Links v0.12.1
		22: BRAT v1.0.2
		23: Auto Note Mover v1.2.3
		24: Callout Manager v1.1.0
		25: Metadata Menu v0.8.7
		26: Hider v1.5.1
		27: Linter v1.26.0
		28: Tracker v1.14.0
		29: Janitor v1.0.7
		30: Footnote Shortcut v0.1.3
		31: Homepage v4.0.4

RECOMMENDATIONS:
	Custom theme and snippets: for cosmetic issues, please first try updating your theme and disabling your snippets. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community theme and snippets.
	Community plugins: for bugs, please first try updating all your plugins to latest. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community plugins.

### Describe the bug

When I link from note A to excalidraw picture B, link A is not shown as backlink of excalidraw picture B.

### Steps to reproduce

1. Create note A
2. Create picture B
3. Link to picture B from note A
4. See the empty (!) backlinks of picture B

### Expected behavior

note A is listed as backlink of picture B

### Additional context

https://www.loom.com/share/74e9c53220ec479d979e32a6818b1219?sid=c99ad986-4bfc-431e-bf49-268126a3c878

**Comment by zsviczian** [comment]
Backlinks are not an Excalidraw feature, they are managed by Obsidian. You might want to have Obsidian reindex your Vault. See rebuild index under 

<img width="1072" alt="image" src="https://github.com/user-attachments/assets/a991767d-b703-4c43-8c46-87f871172014">

That said, since you can see the link in ExcaliBrain, it means that the link is there in the Obsidian Index, since Excalidraw pulls links from DataView, and in turn DataView is monitoring Obsidian Metadata Cache.

**Comment by koenbunders** [comment]
Thanks for the swift response.
Weird... Obsidian is not showing any backlinks for any Excalidraw pictures...

**Comment by zsviczian** [comment]
I see you are using metadata menu... that is not related to the links, but in the past I had people report Obsidian performance issues when Excalidraw and Metadata menu were both enabled (this was due to Metadata menu's indexing logic running slow on the relatively large Excalidraw files. Do you experience any such issues?

The other performance culprit was Minimal Theme... again, do you experience performance issues?

Can there be a filter in Obsidian somewhere based on a tag like `#excalidraw`?

Just going based on the names of plugins, not knowing what they do... I would try disabling supercharged links, then restarting Obsidian to see what is going on.

then next step is to open a test vault, install excalidraw and see if you can reproduce the issue. If yes, interesting... I don't expect that, if not, then you need to disable all other plugins in your vault, restart Obsidian, and check if the issue persists. If it does not, turn on plugins one by one.

**Comment by koenbunders** [comment]
The issue is solved!

**Comment by zsviczian** [comment]
and what was the solution? Maybe others will come across the same and would benefit from a summary of your solution.

**Comment by koenbunders** [comment]
I wasn't sure, but when I looked at it again, I think I found what confused me.
The issue has nothing to do with Excalidraw.
I was confused by unexpected behavior by Obsidian:
When the 'Linked Mentions' is collapsed, the number of linked mentions is not shown.
The number of linked mentions is only shown after 'Linked Mentions' was expanded once

https://www.loom.com/share/130d21631014476f8704c31aa46849c3?sid=56570182-1a1e-4346-ae3f-1b70b8de7e7d.
